web/build.sh

```diff
@@ -0,0 +1,484 @@
+
+    # Update build number if successful
+    echo
+    echo KeymanWeb resources saved under $1
+    echo
+}
+
+#### Build action definitions ####
+
+if builder_start_action configure; then
+  verify_npm_setup
+
+  if ! builder_has_option --skip-minify; then
+    # NPM install is required for the file to be present.
+    if ! [ -f $minifier ];
+    then
+      echo File $minifier does not exist:  have you set the environment variable \$CLOSURECOMPILERPATH?
+      exit 1
+    fi
+  fi
+
+  builder_finish_action success configure
+fi
+
+if builder_start_action clean; then
+  rm -rf "release/"
+  rm -rf "intermediate/"
+  builder_finish_action success clean
+fi
+
+if builder_start_action clean:samples; then
+  rm -f $PREDICTIVE_TEXT_OUTPUT
+
+  builder_finish_action success clean:samples
+fi
+
+if builder_start_action clean:tools; then
+  tools/build.sh clean
+
+  builder_finish_action success clean:tools
+fi
+
+## Build actions
+
+# Adds a simple version 'header' when there's a main engine build product.
+if builder_has_action build:embed || builder_has_action build:web; then
+  echo "${COLOR_PURPLE}Compiling version ${VERSION}${COLOR_RESET}"
+  echo ""
+fi
+
+if builder_start_action build:samples; then
+  echo "Copying samples & test page resources..."
+  cp "${PREDICTIVE_TEXT_SOURCE}" "${PREDICTIVE_TEXT_OUTPUT}"
+
+  builder_finish_action success build:samples;
+fi
+
+### -embed section start
+
+if builder_start_action build:embed; then
+  $compilecmd -b $NODE_SOURCE/tsconfig.embedded.json -v
+
+  assert_exists $INTERMEDIATE/keyman.js
+
+  echo Embedded TypeScript compiled as $INTERMEDIATE/keyman.js
+
+  copy_resources "$INTERMEDIATE"  # Very useful for local testing.
+
+  finish_nominify $EMBEDDED $EMBED_TARGET
+
+  if ! builder_has_option --skip-minify; then
+    # Create our entire embedded compilation results path.
+    if ! [ -d $EMBED_OUTPUT/resources ]; then
+      mkdir -p "$EMBED_OUTPUT/resources"  # Includes base folder, is recursive.
+    fi
+
+    if [ -f "$EMBED_OUTPUT/keyman.js" ]; then
+      rm $EMBED_OUTPUT/keyman.js 2>/dev/null
+    fi
+
+    minify keyman.js $EMBED_OUTPUT SIMPLE_OPTIMIZATIONS
+    assert_exists $EMBED_OUTPUT/keyman.js
+    echo Compiled embedded application saved as $EMBED_OUTPUT/keyman.js
+
+    # Update any changed resources
+    # echo Copy or update resources
+
+    copy_resources "$EMBED_OUTPUT/resources"
+
+    # Update build number if successful
+    echo
+    echo KMEA/KMEI version $VERSION compiled and saved under $EMBED_OUTPUT
+    echo
+  fi
+
+  if builder_has_option --debug; then
+    # We currently have an issue with sourcemaps for minified versions in embedded contexts.
+    # We should use the unminified one instead for now.
+    cp $EMBED_OUTPUT_NO_MINI/keyman.js $EMBED_OUTPUT/keyman.js
+    # Copy the sourcemap.
+    cp $EMBED_OUTPUT_NO_MINI/keyman.js.map $EMBED_OUTPUT/keyman.js.map
+    echo Uncompiled embedded application saved as keyman.js
+  fi
+
+  # TODO:  handle this block somehow.
```

Comment:
- [ ] Sentry symbol uploads must be resolved before merging to `master`.

Reply:
Is this still a TODO for #7831